### ISSUE TITLE
"core" and "data_set_manager" tests can't run separately

### ISSUE THREAD

**Issue by mccalluc** [refinery-platform/refinery-platform#1353]
Commit: f6b186759d0ff5c
### Steps to reproduce

```
$ for TEST in `find . | grep tests.py\$`; do ./manage.py test $TEST && echo "PASS $TEST" >> /tmp/tests.txt || echo "FAIL $TEST" >> /tmp/tests.txt; done
$ cat /tmp/tests.txt
```
### Observed behavior

```
PASS ./analysis_manager/tests.py
PASS ./annotation_server/tests.py
FAIL ./core/tests.py
FAIL ./data_set_manager/tests.py
PASS ./file_server/tests.py
PASS ./file_store/tests.py
PASS ./galaxy_connector/tests.py
PASS ./visualization_manager/tests.py
PASS ./workflow_manager/tests.py
```

Looking at the fuller output, both are failing with `django.contrib.contenttypes.models.DoesNotExist: ContentType matching query does not exist`
### Expected behavior

Tests pass, even when run one-by-one.

**Comment by mccalluc** [comment]
Looking more closely at the stack traces for both failures, they both end with:
```
  File "/vagrant/refinery/core/models.py", line 2184, in _add_user_to_neo4j
    'core.read_dataset'
  File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/guardian/shortcuts.py", line 513, in get_objects_for_group
    permission__codename=codename)
  File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/django/db/models/manager.py", line 151, in get
    return self.get_queryset().get(*args, **kwargs)
  File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/django/db/models/query.py", line 310, in get
    self.model._meta.object_name)
django.contrib.contenttypes.models.DoesNotExist: ContentType matching query does not exist.
```

The reference to models.py is the only part of the stack which is not in site-packages.

(Found some advice at SO that this is usually cause by fixtures referencing models which don't exist, but deleting them didn't change the behavior here, for better or worse.)

**Comment by mccalluc** [comment]
If I change the code to 
```
            lambda ds: ds.uuid, get_objects_for_group(
                ExtendedGroup.objects.public_group(),
                'core.read_dataset_CLEARLY_WRONG'
            )
```
It fails in the same way... So it's having trouble mapping the permissions recorded in the DB to a Model class?

**Comment by scottx611x** [comment]
@mccalluc are you running these test on the vm or locally?

**Comment by mccalluc** [comment]
@scottx611x : This is in the VM... I think that's the right place to run?

**Comment by scottx611x** [comment]
@mccalluc yep you're correct. Maybe we can try to iron this out when we talk about IGV stuff later?

**Comment by mccalluc** [comment]
Scott was able to reproduce the bug, and you can also see the problem in the IDE's test runner, if you try to run just core or data_set_manager. Tests are dependent on the order in which they are run. We believe the underlying problem is the same as #1146, which isn't easy.

**Comment by scottx611x** [comment]
This has been addressed as of commit: 14499395f76566577090e8a523f39adf4e3017f4
